docs/proposals/fast-sync/trust.md

```diff
@@ -0,0 +1,499 @@
+an honest stakeholder, so the whole list must be correct. Verification
+of subsequent tiny blocks is the same.
+
+If all tiny blocks are correct, headers received at step 8 are
+guaranteed to be created by real slot leaders. According to Common
+Prefix property, at least the oldest block must be created by an
+honest stakeholder, hence it must contain the hash of the correct
+snapshot.
+
+#### Data to download
+
+Size of a serialized stakeholder identifier is 30 bytes, size of a
+public key is 66 bytes and size of a simple signature is 66 bytes. In
+the worst case all signatures are proxy signatures and then a tiny
+block contains at most 8640 signatures and public keys and 4320
+stakeholder identifiers. So the size of one tiny block can be
+estimated as 1275000 bytes (considering CBOR overhead) in the worst
+case.
+
+One year contains 73 epochs. So the size of all tiny blocks is
+approximately 93075000 bytes per year, which is approximately 93
+megabytes. Note, however, that
+some [optimizations](#possible-optiomizations) are possible at the
+cost of extra complexity.
+
+It's also necessary to download some full blocks and headers, but
+their number is limited.
+
+#### Local computations
+
+In order to verify a tiny block one needs to check at most 8640
+signatures and hash at most 4320 public keys. For one year it's needed
+to check 630720 signatures and compute 315360 hashes. These
+computations can be performed in less than a second. So the main
+property is the size of data which should be downloaded.
+
+### Possible optimizations
+
+Using some optimizations it's possible to improve some properties at
+the cost of making the design and/or implementation a bit more
+complex. With currently proposed scheme the size of a tiny block in
+the worst case scenario is about **1275** kilobytes
+
+* More compact serialization. Instead of using CBOR for everything, we
+  can serialize tiny blocks more compactly. Then a hash will take 28
+  bytes, a simple signature and a public key will take 64 bytes. It
+  can make a tiny block a bit smaller, approximately **1231**
+  kilobytes (**90** Mb per year).
+* Put less signatures into a tiny block. For instance, if all slots
+  are filled, it's enough to put first `k` signatures into a tiny
+  block. If some slots don't have blocks, it's probably not enough to
+  put only `k` signatures, but `2 · l` is not necessary. More profound
+  security analysis should be done to determine lower and upper
+  boundaries. In the best case, if `k` signatures is always enough and
+  the previous optimization is done, the size of a tiny block can be
+  about **674** kilobytes (**49** Mb per year).
+* Another option is to make `l` much smaller than `k`. It depends on
+  how much trust we want to have. `k = 2160` is quite a lot. It's much
+  greater than number of confirmations after a transaction is
+  considered stable. Probably having `l = 500` will be enough for
```

Comment:
Don't think we can go that far

Reply:
It depends on who we are targeting. If this mechanism is intended to be used only by some services like shops and exchanges and also end users, I don't see anything bad in having small `l`. These services most likely consider a transaction confirmed after less than `500` blocks. For instance, AFAIU, currently exchanges require about 20 confirmations. So they assume that it's impossible to revert more than 20 blocks. Maybe it will be increased in Shelley, but probably not beyond 500 blocks, because 500 blocks is almost 3 hours in the best case.
Full nodes which maintain the system will mostly stay online and they won't need this mechanism probably.
However, the problem is that we should choose `l` suitable for everybody, so even if 100500 services rely on 20 confirmations but there is a service which needs 1000 confirmations, we should either set `l = 1000` or require that service to use full syncing.